### PR TITLE
Feature/user control panel

### PR DESCRIPTION
**Pull request by Southeaven**

### CODE REVIEW

File: cogboard-webapp/src/actions/thunks.js [hunk 1]
```diff
@@ -224,6 +224,12 @@ const deleteCredentialThunk = id =>
     deleteSettingsItem
   );
 
+const updateUserSettingsThunk = user => dispatch => {
+  fetchData(URL.UPDATE_USER_SETTINGS, { method: 'POST', data: user }).catch(
```
[the remark this answers is not in the document]
Comment (Southeaven) [reply cogboard-webapp/src/actions/thunks.js]: Done

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/validation/admins/AdminsValidator.kt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 package com.cognifide.cogboard.config.validation.admins
 
-import com.cognifide.cogboard.config.model.Admins
+import com.cognifide.cogboard.config.model.Admin
```
Comment (szymon-owczarzak) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/validation/admins/AdminsValidator.kt:3]: Rename `AdminsValidator.kt ` to `AdminValidator.kt ` to be consistent

**Review by szymon-owczarzak — CHANGES_REQUESTED** [review]
Please revert all `*.md` file changes

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/validation/admins/AdminsValidator.kt (hunk 2 above)
Comment (yahorm) [reply cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/validation/admins/AdminsValidator.kt]: done

**Comment by yahorm** [comment]
> Please revert all `*.md` file changes

fixed

File: .github/release-drafter.yml [hunk 3]
```diff
@@ -3,12 +3,12 @@ template: |
...
+  - title: "🐛 Bug Fixes"
+    label: "bug"
+  - title: "🧰 Maintenance"
     labels:
-    - 'maintenance'
-    - 'documentation'
-    - 'dependencies'
+      - "maintenance"
+      - "documentation"
+      - "dependencies"
```
Comment (szymon-owczarzak) [inline .github/release-drafter.yml:14]: Please revert this also

File: .travis.yml [hunk 4]
```diff
@@ -12,4 +12,4 @@ before_cache:
 cache:
   directories:
     - "$HOME/.gradle/caches/"
-    - "$HOME/.gradle/wrapper/"
\ No newline at end of file
+    - "$HOME/.gradle/wrapper/"
```
Comment (szymon-owczarzak) [inline .travis.yml]: Please revert this also

File: .github/release-drafter.yml (hunk 3 above)
Comment (yahorm) [reply .github/release-drafter.yml]: done

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/UserService.kt [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        val wrongPassMsg = config.getString("wrongPassMsg") ?: "Please, enter correct Password"
+        val admin = newLoginData.toAdmin()
+        val currentPassword = newLoginData.getString(PROP_CURRENT_PASSWORD)
+        return if (checkCurrentPassword(currentPassword)) {
+            update(admin)
+            response()
+        } else response(wrongPassMsg)
+    }
+
+    private fun checkCurrentPassword(currentPassword: String): Boolean {
```
Comment (szymon-owczarzak) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/UserService.kt:28]: `checkCurrentPassword` is not the best description of what the method is actually doing.
`confirmPassword` maybe or smth like this

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/security/LoginHandler.kt [hunk 6]
```diff
@@ -28,34 +32,36 @@ class LoginHandler(val storage: Storage = VolumeStorageFactory.admins()) : Routi
 
         return Handler { ctx ->
             ctx.bodyAsJson?.let {
-                val username = it.getString("username", "")
-                val password = it.getString("password", "")
+                val username = it.getString(PROP_USERNAME, "")
+                val password = it.getString(PROP_PASSWORD, "")
```
Comment (szymon-owczarzak) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/security/LoginHandler.kt:36]: NIce 👍

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/security/LoginHandler.kt [hunk 7]
```diff
@@ -28,34 +32,36 @@ class LoginHandler(val storage: Storage = VolumeStorageFactory.admins()) : Routi
...
+        return if (username == name) {
+            val password = admin.getString(PROP_PASSWORD)
+            Admin(username, password)
+        } else null
     }
 
-    private fun sendJWT(ctx: RoutingContext, user: String) {
-        ctx.response().end(generateJWT(user))
-    }
+    private fun isNotAuthorized(admin: Admin, password: String?) =
```
Comment (szymon-owczarzak) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/security/LoginHandler.kt:56]: can we revert this - it is always a better practise to use positive methods without negation:
```
private fun isAuthorized(admin: Admin, password: String?) = 
    admin.password.isNotBlank() && admin.password == password
```

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/security/LoginHandler.kt [hunk 8]
```diff
@@ -1,6 +1,10 @@
 package com.cognifide.cogboard.security
 
-import com.cognifide.cogboard.CogboardConstants
+import com.cognifide.cogboard.CogboardConstants.Companion.PROP_PASSWORD
+import com.cognifide.cogboard.CogboardConstants.Companion.PROP_USER
+import com.cognifide.cogboard.CogboardConstants.Companion.PROP_USERNAME
+import com.cognifide.cogboard.CogboardConstants.Companion.STATUS_CODE_401
```
Comment (szymon-owczarzak) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/security/LoginHandler.kt:6]: 👍

File: cogboard-local-compose.yml [hunk 9]
```diff
@@ -22,8 +22,8 @@ services:
       - "./mnt:/data"
     networks:
       - cognet
-  #    ports:
-  #      - "18092:18092"
+#    ports:
+#      - "18092:18092"
```
Comment (szymon-owczarzak) [inline cogboard-local-compose.yml:27]: revert this also

**Review by szymon-owczarzak — DISMISSED** [review]
Please consider my last comments - great work !

File: cogboard-local-compose.yml (hunk 9 above)
Comment (yahorm) [reply cogboard-local-compose.yml]: fixed

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/security/LoginHandler.kt (hunk 7 above)
Comment (yahorm) [reply cogboard-app/src/main/kotlin/com/cognifide/cogboard/security/LoginHandler.kt]: fixed

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/UserService.kt (hunk 5 above)
Comment (yahorm) [reply cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/UserService.kt]: done

**Review by szymon-owczarzak — DISMISSED** [review]

**Review by GrzegorzBednarski — APPROVED** [review]

**Pull request merged**